File: programmers/43165.py
# 출처 : https://programmers.co.kr/learn/courses/30/lessons/43165
def solution(numbers, target):
    answer = []

    def dfs(level, sum, answer):
        if(level == len(numbers)):
            if(sum == target):
                answer.append(1)
            return

        dfs(level+1, sum+numbers[level], answer)
        dfs(level+1, sum-numbers[level], answer)

    dfs(0, 0, answer)

    return len(answer)

File: programmers/43165.js
function solution(numbers, target) {
    var answer = 0;

    function dfs(level, sum) {
        if (level == numbers.length) {
            if (sum == target) {
                answer++;
            }
            return
        }

        dfs(level + 1, sum + numbers[level]);
        dfs(level + 1, sum - numbers[level]);

    }

    dfs(0, 0);
    return answer;
}

solution([1, 1, 1, 1, 1], 3)